src/s3/util/s3Tab.ts

```diff
@@ -0,0 +1,103 @@
+/*!
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+//import { Bucket, DownloadFileRequest, File, S3Client } from '../../shared/clients/s3Client'
+
+//need to subscribe to ondidsave and upload to s3, might need the S3FileNode
+
+export class S3Tab {
+    //private file: File
+    private s3Uri: vscode.Uri
+    private window: typeof vscode.window
+    private editor: vscode.TextEditor | undefined
+
+    //private editing: boolean
+
+    public constructor(private fileUri: vscode.Uri, window = vscode.window) {
+        this.s3Uri = vscode.Uri.parse('s3:' + this.fileUri.fsPath)
+        this.window = window
+    }
+
+    public async openFileOnReadOnly(workspace = vscode.workspace): Promise<vscode.TextEditor | undefined> {
```

Comment:
Does this need to return anything? You're not doing anything with the values now; can this be used for tests?

Reply:
That was the main reason, but ended up testing other way